src/shared/sleep-config.c

```diff
@@ -102,6 +102,39 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+int battery_is_low(void) {
+        _cleanup_free_ char *bat_cap_level = NULL;
+        int r;
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/capacity_level", &bat_cap_level);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/capacity_level: %m");
+
+        log_debug("Current battery level: %s", bat_cap_level);
+
+        return STR_IN_SET(bat_cap_level, "Critical", "Low");
+}
```

Comment:
there are laptops with multiple batteries, we need to do something reasonable with them. i.e. enumerate all of them, and only consider the battery low if all of them are low or so.

Reply:
As mentioned above, this really should be a follow-up, in the vast majority of cases there's only one, so it's fine to start with this and build on it later